### ISSUE TITLE
Add support for verifying content downloaded from mathjax_path with the integrity html attribute

### ISSUE THREAD

**Issue by stxlvt** [sphinx-doc/sphinx#3784]
In order to guard web users from compromised CDN servers (which is a frightening concept), Mozilla and Chrome has added support for the browser to [verify the integrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity) of scripts and/or css used in a webpage by adding one or more space-separated hash digests to the `integrity` attribute of the `<script>` tag, like so:

    <script src="https://example.com/example-framework.js"
        integrity="sha384-oqVuAfXRKap7fdgcCY5uykM6+R9GqQ8K/uxy9rx7HNQlGYl1kPzQho1wx4JwY8wC"
        crossorigin="anonymous"></script>

I suggest adding a `mathjax_integrity` configuration value.

If set it must be a string like `"sha384-oqVuAfXRKap7fdgcCY5uykM6+R9GqQ8K/uxy9rx7HNQlGYl1kPzQho1wx4JwY8wC"`, and this string, in quotes, will then be prepended with `integrity=` and added as an attribute to the mathjax script tag.

If `mathjax_integrity` it is not set then no `integrity` attribute will be added and things will work exactly as before.

**Comment by tk0miya** [comment]
Besides this, there are some options for `<script>` tag. So I prefer to add `mathjax_options`:
```
mathjax_options = {
    'integrity': 'sh384-...',
    'crossorigin': 'anonymous',
}
```

**Comment by tk0miya** [comment]
Fixed by #5084 
Thank you for reporting.
